flow/project.py

```diff
@@ -291,13 +304,14 @@ def __init__(self, id, name, job, cmd, directives=None):
         self.directives._keys_set_by_user = keys_set_by_user
 
     def __str__(self):
-        return "{}({})".format(self.name, self.job)
+        assert len(self._jobs) == 1
+        return f"{self.name}({str(self._jobs[0])})"
```

Comment:
This is an internal class we don't need the `len == 1` assumption. We could use this in `JobOperation` though.

Reply:
After we include actual aggregation, this will have to change to something like.
`f"{self.name}-{len(self._jobs)}: ({str(self._jobs[0])}-{str(self._jobs[-1])})"` for every aggregate (even for length 1).

Please see this [comment](https://github.com/glotzerlab/signac-flow/pull/324#discussion_r467830066) by @csadorf

Reply:
I am aware. I more mean, we can do that now given this is an internal class now. We just need this behavior for `JobOperation`

Reply:
@b-butler If we do this, I'll have to change the template-reference-data twice(once here and another in #334 )
This will result in a lot of additional work.
I think this should be addressed in #334

Reply:
The assert should be in place wherever we actually make that assumption. Once that assumption is removed (e.g. in #334), the asserts will automatically alert us to spots in the code that require a revised implementation.